ui/desktop/src/components/ReportFailureModal.tsx

```diff
@@ -0,0 +1,285 @@
+import { useState, useEffect, useCallback } from 'react';
+import { Button } from './ui/button';
+import { Textarea } from './ui/textarea';
+import { Card, CardContent } from './ui/card';
+import { AlertCircle, CheckCircle, Loader2 } from 'lucide-react';
+import Modal from './Modal';
+import { getSecretKey } from '../config';
+import { getExtensions, providers } from '../api/sdk.gen';
+
+interface SystemInfo {
+  gooseVersion: string;
+  osVersion: string;
+  platform: string;
+  architecture: string;
+  providerType?: string;
+  extensionCount: number;
+}
+
+interface ReportFailureModalProps {
+  isOpen: boolean;
+  onClose: () => void;
+}
+
+export default function ReportFailureModal({ isOpen, onClose }: ReportFailureModalProps) {
+  const [description, setDescription] = useState('');
+  const [systemInfo, setSystemInfo] = useState<SystemInfo | null>(null);
+  const [recentErrors, setRecentErrors] = useState<string[]>([]);
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const [submitStatus, setSubmitStatus] = useState<'idle' | 'success' | 'error'>('idle');
+
+  const getExtensionCount = async (): Promise<number> => {
+    try {
+      const secretKey = getSecretKey();
+      const response = await getExtensions({
+        headers: secretKey ? { 'X-Secret-Key': secretKey } : {},
```

Comment:
you should not to include the secret key here, as it's globally configured for the generated client sdk